app/controllers/foreman_puppet_enc/config_groups_controller.rb

```diff
@@ -0,0 +1,43 @@
+module ForemanPuppetEnc
+  class ConfigGroupsController < ::ApplicationController
+    include Foreman::Controller::AutoCompleteSearch
+    include Foreman::Controller::Parameters::ConfigGroup
+
+    before_action :find_resource, only: %i[edit update destroy]
+
+    def index
+      @config_groups = resource_base_search_and_page
+    end
+
+    def new
+      @config_group = ConfigGroup.new
```

Comment:
This is not necessary, but it would be safer to do so.
```suggestion
      @config_group = ForemanPuppetEnc::ConfigGroup.new
```